Remove claim about semicolons from docs (#12944)
eslint/eslint: docs/rules/semi.md

# require or disallow semicolons instead of ASI (semi)

> JavaScript doesn't require semicolons at the end of each statement. In many cases, the JavaScript engine can determine that a semicolon should be in a certain spot and will automatically add it. This feature is known as **automatic semicolon insertion (ASI)** and is considered one of the more controversial features of JavaScript. For example, the following lines are both valid:

Reviewer: Personally, I'd prefer to phrase it as
```suggestion
JavaScript requires semicolons so much that it will insert one for you if you omit it, wherever it can determine that a semicolon should be. This feature is known as **automatic semicolon insertion (ASI)** and is considered one of the more controversial features of JavaScript. For example, the following lines are both valid:
```
:-p
